server/src/main/antlr3/org/apache/iotdb/db/sql/parse/TqlLexer.g

```diff
@@ -0,0 +1,539 @@
+    ;
+
+K_DATATYPE
+    : D A T A T Y P E
+    ;
+
+K_ENCODING
+    : E N C O D I N G
+    ;
+
+K_COMPRESSOR
+    : C O M P R E S S O R
+    ;
+
+K_STORAGE
+    : S T O R A G E
+    ;
+
+K_TO
+    : T O
+    ;
+
+K_PROPERTY
+    : P R O P E R T Y
+    ;
+
+K_LABEL
+    : L A B E L
+    ;
+
+
+K_LINK
+    : L I N K
+    ;
+
+K_UNLINK
+    : U N L I N K
+    ;
+
+K_SHOW
+    : S H O W
+    ;
+
+K_METADATA
+    : M E T A D A T A
+    ;
+
+K_DESCRIBE
+    : D E S C R I B E
+    ;
+
+K_INDEX
+    : I N D E X
+    ;
+
+K_ON
+    : O N
+    ;
+
+K_USING
+    : U S I N G
+    ;
+
+K_DROP
+    : D R O P
+    ;
+
+K_MERGE
+    : M E R G E
+    ;
+
+K_LIST
+    : L I S T
+    ;
+
+K_USER
+    : U S E R
+    ;
+
+K_PRIVILEGES
+     : P R I V I L E G E S
+     ;
+
+K_ROLE
+    : R O L E
+    ;
+
+K_ALL
+    : A L L
+    ;
+
+K_OF
+    : O F
+    ;
+
+K_QUIT
+    : Q U I T
+    ;
+
+K_ALTER
+    : A L T E R
+    ;
+
+K_PASSWORD
+    : P A S S W O R D
+    ;
+
+K_GRANT
+    : G R A N T
+    ;
+
+K_REVOKE
+    : R E V O K E
+    ;
+
+K_PATH
+    : P A T H
+    ;
+
+K_LOAD
+    : L O A D
+    ;
+
+K_WATERMARK_EMBEDDING
+    : W A T E R M A R K '_' E M B E D D I N G
+    ;
+
+//************** logical operator***********
+OPERATOR_AND
+    : A N D
+    | '&'
+    | '&&'
+    ;
+
+OPERATOR_OR
+    : O R
+    | '|'
+    | '||'
+    ;
+
+OPERATOR_NOT
+    : N O T | '!'
+    ;
+
+
+//**************** data type ***************
+K_PLAIN
+	: P L A I N
+	;
+K_PLAIN_DICTIONARY
+	: P L A I N '_' D I C T I O N A R Y
+	;
+K_RLE
+	: R L E
+	;
+K_DIFF
+	: D I F F
+	;
+K_TS_2DIFF
+	: T S '_' '2' D I F F
+	;
+K_BITMAP
+	: B I T M A P
+	;
+K_GORILLA
+	: G O R I L L A
+	;
+K_REGULAR
+	: R E G U L A R
+	;
+
+K_ADD
+    : A D D
+    ;
+// *************** comparison *******
+OPERATOR_GT
+    : '>'
+    ;
+
+OPERATOR_GTE
+    : '>='
+    ;
+
+OPERATOR_LT
+    : '<'
+    ;
+
+OPERATOR_LTE
+    : '<='
+    ;
+
+OPERATOR_EQ
+    : '=' | '=='
+    ;
+
+OPERATOR_NEQ
+    : '!=' | '<>'
+    ;
+
+//************ operator *******
+STAR
+    : '*'
+    ;
+
+MINUS
+    : '-'
+    ;
+
+PLUS
+    : '+'
+    ;
+
+DIVIDE
+    : '/'
+    ;
+
+//**************** symbol ***************
+SEMI
+    : ';'
+    ;
+
+DOT
+    : '.'
+    ;
+
+COMMA
+    : ','
+    ;
+
+LR_BRACKET
+    : '('
+    ;
+
+RR_BRACKET
+    : ')'
+    ;
+
+LS_BRACKET
+    : '['
+    ;
+
+RS_BRACKET
+    : ']'
+    ;
+
+STRING_LITERAL
+   : DQUOTA_STRING
+   | SQUOTA_STRING
+   ;
+
+fragment DQUOTA_STRING
+   : '"' ( '\\'. | '""' | ~('"'| '\\') )* '"'
+   ;
+
+fragment SQUOTA_STRING
+   : '\'' ('\\'. | '\'\'' | ~('\'' | '\\'))* '\''
+   ;
+
+DATETIME
+    : INT ('-'|'/') INT ('-'|'/') INT
+      (T | WS)
+      INT ':' INT ':' INT (DOT INT)?
+      (('+' | '-') INT ':' INT)?
+    ;
+
+EXPONENT : INT ('e'|'E') ('+'|'-')? INT ;
+
+    
+ID
+    :	('a'..'z'|'A'..'Z'|'_') ('a'..'z'|'A'..'Z'|'0'..'9'|'_'|'-')*
+    ;
+
+INT
+    : NUM+
+    ;
+
+fragment
+NUM
+    :   '0'..'9'
+    ;
+
+
+
+
+// ***************************
+fragment A
```

Comment:
Why don’t you put them together?
```
fragment
Letter
    : 'a'..'z' | 'A'..'Z'
    ;
```

Reply:
Separating them makes it easier to define the other key words, easier to make the sql case insensitive. For example, 
K_SELECT
    : S E L E C T
    ;

If I only define Letter, it will be 
K_SELECT
    : 'SELECT' | 'select'
    ;

Reply:
any questions? @Genius-pig

Reply:
You can use code to uppercase input String, what @Ring-k has done will lead to Antlr file is too long.
Check out what has spark done. https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ParseDriver.scala
https://github.com/apache/spark/blob/master/sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

Reply:
It seems that before parsing, they capitalize all characters in a string, and while it is not case-sensitive to keywords, it is also case-insensitive to paths, values, and so on. To aviod it, I chose the way as it is now. There are some other examples.
https://github.com/antlr/grammars-v3/blob/master/mysql/MySQL.g

I think both ways of implementation are fine. If you think the ANTLR file is too long, separating lexer and parser rules in different files is a solution, as what we have done.

Reply:
I agree with you. There are both ways to handle case-insensitive according to https://github.com/antlr/antlr4/blob/master/doc/case-insensitive-lexing.md which is the Antlr official site. Although your way may cause a little slower, it can handle both case-sensitive and case-insensitive keywords.